field.cpp

```diff
@@ -385,6 +477,41 @@ std::vector<GF2E> build_from_roots(const std::vector<GF2E> &roots) {
   poly[len] = GF2E(1);
   return poly;
 }
+// normal eval precomputation
+std::vector<GF2E> eval_precompute(const GF2E &point, size_t poly_size) {
+  std::vector<GF2E> out;
+  out.reserve(poly_size);
+
+  GF2E temp = point;
+  out.push_back(temp);
+  for (size_t i = 1; i < poly_size; ++i) {
+    temp *= point;
+    out.push_back(temp);
+  }
+  return out;
+}
+
+// normal optmized polynomial evaluation with precomputation optmization
+GF2E eval_fast(const std::vector<GF2E> &poly, const std::vector<GF2E> &x_pow_n,
+               const size_t lambda) {
+  __m128i acc = _mm_set_epi64x(0, poly[0].get_data());
```

Comment:
Can this not just call dot_product? Then we don't need get_data, and lamda at all.